crates/ide/src/goto_definition.rs

```diff
@@ -30,49 +36,94 @@ pub(crate) fn goto_definition(
     let original_token = pick_best(file.token_at_offset(position.offset))?;
     let token = sema.descend_into_macros(original_token.clone());
     let parent = token.parent();
+    if let Some(comment) = ast::Comment::cast(token.clone()) {
+        let (link_name, ns) =
+            extract_positioned_link_from_comment(position, comment, file.clone(), token.clone())?;
+        let nav = nav_target_from_positioned_link(db, sema, parent, &link_name, ns)?;
+        return Some(RangeInfo::new(original_token.text_range(), vec![nav]));
+    }
+
+    let nav_targets = nav_targets_from_node(&sema, position, parent)?;
+
+    Some(RangeInfo::new(original_token.text_range(), nav_targets))
+}
+
+fn extract_positioned_link_from_comment(
+    position: FilePosition,
+    comment: ast::Comment,
+    file: SyntaxNode,
+    token: SyntaxToken,
+) -> Option<(String, Option<hir::Namespace>)> {
+    let comment_range = token.text_range();
+    if let Some(doc_comment) = comment.doc_comment() {
+        let mut min_distance = file.text_range().end();
+        let mut positioned_link = None;
+        let def_links = extract_definitions_from_markdown(doc_comment);
+        for (link, ns, def_link_range) in def_links {
+            let matched_position =
+                comment_range.start() + TextSize::from(def_link_range.start as u32);
+            if let Some(distance) = position.offset.checked_sub(matched_position) {
+                if let Some(_) = min_distance.checked_sub(distance) {
+                    min_distance = distance;
+                    positioned_link = Some((link, ns))
+                };
+            };
+        }
+        positioned_link
+    } else {
+        None
+    }
+}
 
-    let nav_targets = match_ast! {
+fn nav_target_from_positioned_link(
```

Comment:
I don't think it makes sense to extract this function, it is just the common case for `goto_definition`